provisioner/registryv1/controllers/bundleinstance_controller.go

```diff
@@ -93,21 +96,19 @@ func (r *BundleInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	bi := &olmv1alpha1.BundleInstance{}
 	if err := r.Get(ctx, req.NamespacedName, bi); err != nil {
-		if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		return ctrl.Result{}, err
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	defer func() {
 		bi := bi.DeepCopy()
 		bi.ObjectMeta.ManagedFields = nil
-		if err := r.Status().Patch(ctx, bi, client.Apply, client.FieldOwner("core.rukpak.io/registry+v1")); err != nil {
+		if err := r.Status().Patch(ctx, bi, client.Apply, client.FieldOwner(registryV1ProvisionerID)); err != nil {
 			l.Error(err, "failed to patch status")
 		}
 	}()
 
 	b := &olmv1alpha1.Bundle{}
 	if err := r.Get(ctx, types.NamespacedName{Name: bi.Spec.BundleName}, b); err != nil {
+		bi.Status.InstalledBundleName = ""
```

Comment:
Just a quick note: we return an error whenever we encounter a Bundle lookup failure, which means we consistently requeue until we reach the backoff limit. Does it make more sense to return `client.IgnoreNotFound(err)` so we differential a Bundle that doesn't exist (at least in the context of the current cache element) vs. a different error type.

Reply:
Another implementation could be catching the delete event and resetting this status.InstalledBundleName field.